### PR TITLE
1351 select positioning bugs

### PR DESCRIPTION
**Pull request by benjamincharity**
- Fix horizontal panel position for checkbox and optgroup selects
- Re-enable panel opening up
- Fix issue where panel opens upwards too far away from select
- Selected option should be vertically centered over select

### CODE REVIEW

File: demo/app/components/select/select.component.html [hunk 1]
```diff
@@ -76,15 +87,17 @@ <h4 tsSelectOptionDisplay>None</h4>
...
       >
-        {{ option.slug }}
         <!--
-           -<ng-template let-option>
-           -  <div class="myClass">
-           -    <h4 tsSelectOptionDisplay>{{ option?.foo }}</h4>
-           -    <small>{{ option?.slug }}</small>
-           -  </div>
-           -</ng-template>
+           -{{ option.slug }}
```
[the remark this answers is not in the document]
Comment (benjamincharity) [reply demo/app/components/select/select.component.html]: Leaving this in for easy switching for the demos. We are constantly needing to check this select with a single value and with a template. This allows us to just comment/uncomment one or the other without rewriting it every time we touch the select.

File: terminus-ui/select/src/select.component.scss [hunk 2]
```diff
@@ -113,11 +113,35 @@ $ts-select-placeholder-arrow-space: 2 * ($ts-select-arrow-size + $ts-select-arro
...
 
+// Position the filter input inside the panel
+.cdk-overlay-pane {
+  .ts-select-panel__filter-input {
+    background-color: color(pure);
+    border-bottom: 1px solid color(utility, light);
+    left: 0;
+    padding: 8px;
+    position: absolute;
+    right: -32px;
```
[the remark this answers is not in the document]
Comment (benjamincharity) [reply terminus-ui/select/src/select.component.scss]: Unfortunately, no. It's kind of a magic number for this select that doesn't exist in our spacing map.

File: terminus-ui/select/src/select-animations.ts [hunk 3]
```diff
@@ -31,12 +31,12 @@ export const tsSelectAnimations: {
...
     state('showing', style({
       opacity: 1,
-      minWidth: 'calc(100% + 32px)', // 32px = 2 * 16px padding
+      minWidth: 'calc(100% + 32px)',
       transform: 'scaleY(1)',
     })),
     state('showing-multiple', style({
       opacity: 1,
-      minWidth: 'calc(100% + 64px)', // 64px = 48px padding on the left + 16px padding on the right
+      minWidth: 'calc(100% + 24px)',
```
Comment (atlwendy) [inline terminus-ui/select/src/select-animations.ts]: curious... where does this 24 comes from?

Comment (benjamincharity) [reply terminus-ui/select/src/select-animations.ts]: I believe it's a combination of 16px and 8px which is 1em + padding. The code was originally copied from material, but that math in the old comment didn't match our new layout

**Pull request merged**